pandas/core/internals/array_manager.py

```diff
@@ -173,18 +178,48 @@ def _verify_integrity(self) -> None:
     def reduce(
         self: T, func: Callable, ignore_failures: bool = False
     ) -> Tuple[T, np.ndarray]:
-        # TODO this still fails because `func` assumes to work on 2D arrays
-        # TODO implement ignore_failures
-        assert self.ndim == 2
+        """
+        Apply reduction function column-wise, returning a single-row ArrayManager.
 
-        res_arrays = []
-        for arr in self.arrays:
-            res = func(arr, axis=0)
-            res_arrays.append(np.array([res]))
+        Parameters
+        ----------
+        func : reduction function
+        ignore_failures : bool, default False
+            Whether to drop columns where func raises TypeError.
+
+        Returns
+        -------
+        ArrayManager
+        np.ndarray
+            Indexer of column indices that are retained.
+        """
+        result_arrays: List[np.ndarray] = []
+        result_indices: List[int] = []
+        for i, arr in enumerate(self.arrays):
+            try:
+                res = func(arr, axis=0)
+            except TypeError:
+                if not ignore_failures:
+                    raise
+            else:
+                # TODO NaT doesn't preserve dtype, so we need to ensure to create
+                # a timedelta result array if original was timedelta
+                # what if datetime results in timedelta? (eg std)
```

Comment:
can't you simply use na_value_for_dtype and *always* do this?

Reply:
Then we need to know the output dtype of the reduction (which is not necessarily the same as the input dtype, and something that is currently not known here). 

I suppose that in principle for the built-in (non-object) dtypes we can write up a "(in_dtype, reduction) -> out_dtype" table. 
We already do something similar for groupby as well, actually (but specifically for the nullable dtypes):

https://github.com/pandas-dev/pandas/blob/83479e1efc3cdbe31d8042eebaed8a6a90bb3b1c/pandas/core/dtypes/cast.py#L367-L397

Reply:
right, yeah i think we had an issue to move this more generally array ops. its basically take this output and create the correct dtype, using the input if its warranted (e.g. with all-NaN). so prob worth moving / formalizing this.

Reply:
@jreback are you OK with deferring this to a follow-up issue / PR (this PR already implements `ignore_failures` and enables tests, so it useful on its own without solving *everything*)